stdlib/streams/src/main/ballerina/streams/snapshot.bal

```diff
@@ -0,0 +1,350 @@
+    }
+    return wch;
+}
+
+# Function to close writable/readable character channels.
+# + c - Character channel to be closed.
+function closeCharChannel(any c) {
+    if (c is io:ReadableCharacterChannel) {
+        var err = c.close();
+    } else if (c is io:WritableCharacterChannel) {
+        var err = c.close();
+    }
+}
+
+# Function to read a text file.
+# + filePath - A `string` path to the file.
+# + encoding - A `string` indicating the encoding type.
+# + return - The `string` content of the file.
+function readFile(string filePath, string encoding) returns string {
+    string content = "";
+    io:ReadableCharacterChannel? rch = createReadableCharacterChannel(filePath, encoding);
+    if (rch is io:ReadableCharacterChannel) {
+        string|error? c = readAllCharacters(rch);
+        closeCharChannel(rch);
+        if (c is string) {
+            content = c;
+        }
+    }
+    return content;
+}
+
+# Function to write text to a file.
+# + filePath - A `string` path to the file.
+# + encoding - A `string` indicating the encoding type.
+# + content - A `string` content to be written.
+# + return - An `int` indicating how many chars got written.
+function writeToFile(string filePath, string encoding, string content) returns int {
+    int written = 0;
+    io:WritableCharacterChannel? wch = createWritableCharacterChannel(filePath, encoding);
+    if (wch is io:WritableCharacterChannel) {
+        int|error? i = writeCharacters(wch, content, 0);
+        closeCharChannel(wch);
+        if (i is int) {
+            written = i;
+        }
+    }
+    return written;
+}
+
+# Function to serialize a Snapshotable state and write that to given file.
+# + persistancePath - A `string` path of the persistance directory.
+# + return - An `error` if state cannot be written to a file.
+function writeStateToFile(string persistancePath) returns error? {
+    string? snapshotFile = ();
+    time:Time ct = time:currentTime();
+    int currentTimeMillis = ct.time;
+    internal:Path pPath = new(persistancePath);
+    if (!pPath.exists()) {
+        error? e = pPath.createDirectory();
+    }
+    internal:Path path = pPath.resolve(string.convert(currentTimeMillis));
+    if (!path.exists()) {
+        error? e = path.createFile();
+        if (e is ()) {
+            snapshotFile = path.getPathValue();
+        }
+    }
+    if (snapshotFile is string) {
+        string serialized = serialize(streamsPersistanceState);
+        int written = writeToFile(snapshotFile, "UTF8", serialized);
+        if (written > 0) {
+            return;
+        } else {
+            error e = error("Error while writing streaming state to a file.");
+            return e;
+        }
+    } else {
+        error e = error("Error while creating snapshot file: " + path.getPathValue());
+        return e;
+    }
+}
+
+# Function to log a persitance error.
+# + error - The `error` occured while persisting the state.
+function persitanceError(error e) {
```

Comment:
`persitance` -> `persistence`? Applicable elsewhere too.

Reply:
Fixed in d345721